src/main/java/com/basho/riak/client/core/converters/TimeSeriesPBConverter.java

```diff
@@ -72,26 +85,28 @@ public static QueryResult convertPbQueryResp(RiakKvPB.TsQueryResp response)
 
     public static Collection<RiakKvPB.TsRow> convertRowsToPb(Collection<Row> rows)
     {
-        ArrayList<RiakKvPB.TsRow> pbRows = new ArrayList<RiakKvPB.TsRow>(rows.size());
+        final ArrayList<RiakKvPB.TsRow> pbRows = new ArrayList<RiakKvPB.TsRow>(rows.size());
 
         for (Row row : rows)
         {
-            pbRows.add(convertRowToPb(row));
+            final RiakKvPB.TsRow.Builder rowBuilder = RiakKvPB.TsRow.newBuilder();
```

Comment:
Looks like we may be able to do that with the clear message. 

https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/Message.Builder#clear()

Just need to make it thread safe :)